scheduler/plugin/plugin.go

```diff
@@ -191,10 +180,7 @@ var _ framework.FilterPlugin = (*wasmPlugin)(nil)
 
 // Filter implements the same method as documented on framework.FilterPlugin.
 func (pl *wasmPlugin) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
-	pl.pool.assignedToSchedulingPodLock.Lock()
-	defer pl.pool.assignedToSchedulingPodLock.Unlock()
-
-	g, err := pl.pool.getOrCreateGuest(ctx, pod.GetUID())
+	g, err := pl.pool.getForScheduling(ctx, cycleID(pod))
```

Comment:
`assignedToSchedulingPodLock` was the lock to protect guest to be used in parallel. (because Filter can be called in parallel during preemption)
And here, because you remove `assignedToSchedulingPodLock`, the guest getting here can be used in parallel during preemption, right? maybe we should keep `assignedToSchedulingPodLock`?

Reply:
I see. indeed the pool lock won't prevent overlapping calls to use the guest.

So I think I need to revise the docs because scheduling is also not one at a time.

Reply:
I think state problem is too abstract for me to have brought up, yet because we don't implement it in this plugin, yet.

I will fix the overlapping problem on this PR and bear in mind the details about state later!

Reply:
I changed the impl to serialize access during scheduling, and put some notes about my understanding of it. Thanks for all the help and feel free to correct more